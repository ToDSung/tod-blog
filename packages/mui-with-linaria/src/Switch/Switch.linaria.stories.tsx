import { useArgs } from '@storybook/preview-api';
import type { Meta, StoryObj } from '@storybook/react';

import SwitchLinaria from './Switch.linaria';

const meta = {
  title: 'mui-with-linaria/SwitchLinaria',
  component: SwitchLinaria,
} satisfies Meta<typeof SwitchLinaria>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    checked: true,

    children: 'Switch Label',
  },
  render: args => {
    const [{ checked }, updateArgs] = useArgs();

    const onChange = () => {
      updateArgs({ checked: !checked });
    };

    return <SwitchLinaria {...args} onChange={onChange}></SwitchLinaria>;
  },
};
