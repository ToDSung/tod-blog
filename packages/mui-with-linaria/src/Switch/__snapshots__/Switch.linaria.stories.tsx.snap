// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mui-with-linaria/SwitchLinaria Default smoke-test 1`] = `
<div class="atm_9s_1txwivl atm_h_1h6ojuz atm_cx_1y44olf">
  <span class="MuiSwitch-root MuiSwitch-sizeMedium atm_1j4sjcm_hy6zze atm_1myhon8_1vzrvsl atm_141ft4g_1j28jx2 atm_141ft4g_hy6zze css-julti5-MuiSwitch-root">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase">
      <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
             type="checkbox"
             checked
      >
      <span class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb">
      </span>
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </span>
    <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
    </span>
  </span>
  <label>
    Switch Label
  </label>
</div>
`;
